### PR TITLE
[14.0][ADD] partner_stage: add lifecycle stages to Contacts

### PR DESCRIPTION
**Pull request by dreispt**
.

### CODE REVIEW

File: partner_stage/models/res_partner_stage.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Copyright 2021 Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class PartnerStage(models.Model):
+    _name = "res.partner.stage"
+    _description = "Product Stage"
```
Comment (patrickrwilson) [inline partner_stage/models/res_partner_stage.py]: Change Product to Partner.

File: partner_stage/tests/test_partner_stage.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2021 Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.exceptions import ValidationError
+from odoo.tests.common import SavepointCase
+
+
+class TestProductState(SavepointCase):
```
Comment (patrickrwilson) [inline partner_stage/tests/test_partner_stage.py]: Change 'Product' to 'Partner' and fix error that shows in travis.

File: partner_stage/views/res_partner_stage_views.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                    <field name="description" />
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+    <record id="view_partner_stage_tree" model="ir.ui.view">
+        <field name="model">res.partner.stage</field>
+        <field name="arch" type="xml">
+            <tree string="product.state">
```
Comment (patrickrwilson) [inline partner_stage/views/res_partner_stage_views.xml]: Change from 'product' to 'partner'

File: partner_stage/views/res_partner_stage_views.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        </field>
+    </record>
+
+    <record model="ir.actions.act_window" id="action_partner_stage">
+        <field name="name">Stage</field>
+        <field name="res_model">res.partner.stage</field>
+        <field name="view_mode">tree,form</field>
+    </record>
+
+    <record model="ir.ui.menu" id="menu_product_state">
```
Comment (patrickrwilson) [inline partner_stage/views/res_partner_stage_views.xml]: Change from 'product' to 'partner'

**Review by patrickrwilson — CHANGES_REQUESTED** [review]
Tested and works well but needs some changes and travis needs fixed.

**Comment by dreispt** [comment]
@patrickrwilson Fixed!

**Comment by dreispt** [comment]
Added "state", plus corresponding migration scrips from previous version.

**Review by vongoethe — APPROVED** [review]
Works great!

**Review by rafaelbn — APPROVED** [review]

**Pull request merged**